promql/histogram_stats_iterator.go

```diff
@@ -19,32 +19,39 @@ import (
 	"github.com/prometheus/prometheus/tsdb/chunkenc"
 )
 
-type histogramStatsIterator struct {
+type HistogramStatsIterator struct {
```

Comment:
Now that this is exported, better give it a meaningful doc comment. I would probably recycle most of the stuff that was used for `NewHistogramStatsIterator` below and make the latter much shorter.

Reply:
Done in 1141530.